client/client.go

```diff
@@ -259,8 +260,30 @@ func (c *Cmd) Dial() error {
 	if err := c.UserKeyConfig(); err != nil {
 		return err
 	}
-	addr := net.JoinHostPort(c.HostName, c.Port)
-	cl, err := ssh.Dial(c.network, addr, &c.config)
+	// Sadly, no vsock in net package.
+	var (
+		cl   *ssh.Client
+		err  error
+		addr string
+	)
+
+	switch c.network {
+	case "vsock":
+		id, port, err := vsockIdPort(c.HostName, c.Port)
+		conn, err := vsock.Dial(id, port, nil)
+		if err != nil {
+			return err
+		}
+		addr = fmt.Sprintf("%#x:%d", id, port)
+		c, chans, reqs, err := ssh.NewClientConn(conn, addr, &c.config)
```

Comment:
i'm not familiar with the ssh package, but would it be worth having all cases (vsock, inet) operate on chans and addrs, and then call NewClientConn and NewClient outside the switch statement?

Reply:
yeah, that's a great idea, see what you think of this now.